From 2ffb468a41d28eceda3afe869dd8c5af70203bf4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 19 Dec 2018 17:19:27 +0100
Subject: [PATCH] Fix importing PNG files with transparency

Pillow does not allow to save images with transparency to JPEG. These images need to be converted first.
---
 paperwork-backend/paperwork_backend/img/page.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/paperwork-backend/paperwork_backend/img/page.py b/paperwork-backend/paperwork_backend/img/page.py
index ddbb9214..86d5e4be 100644
--- a/paperwork-backend/paperwork_backend/img/page.py
+++ b/paperwork-backend/paperwork_backend/img/page.py
@@ -141,6 +141,7 @@ class ImgPage(BasicPage):
 
     def __set_img(self, img):
         with self.fs.open(self.__img_path, 'wb') as fd:
+            img = img.convert("RGB")
             img.save(fd, format="JPEG")
 
     img = property(__get_img, __set_img)
-- 
2.20.0

